roboteam_ai/src/control/positionControllers/BallHandlePosControl.cpp

```diff
@@ -0,0 +1,514 @@
+    if (ball->visible) {
+        compensationAngle = robotToBall.toAngle() - robot->angle;
+        compensationVelocity = (robot->angle + M_PI_2).toVector2(compensationAngle);
+    }
+
+    switch (rotateStrategy) {
+    case rotateAroundBall: {
+        double maxV = maxForwardsVelocity;
+        double targetVel = deltaAngle.getAngle() > maxV ? maxV
+                                                        : deltaAngle.getAngle() < - maxV ? - maxV :
+                                                          deltaAngle.getAngle();
+
+        robotCommand.vel = robotToBall.rotate(- M_PI_2).stretchToLength(targetVel) - previousVelocity*0.2;
+        robotCommand.angle = robotToBall.toAngle();
+        robotCommand.dribbler = 0;
+        return limitCommand(robotCommand);
+    }
+    case rotateAroundRobot: {
+        int direction = targetAngle - robot->angle > 0.0 ? 1 : - 1;
+        robotCommand.angle = Angle(robot->angle + maxAngularVelocity*direction);
+        robotCommand.dribbler = 1;
+        return limitCommand(robotCommand);
+    }
+    }
+    if (Constants::SHOW_BALL_HANDLE_DEBUG_INFO()) {
+        std::cout << "rotate case not handled" << std::endl;
+    }
+    return {};
+}
+
+RobotCommand BallHandlePosControl::travelWithBall(TravelStrategy travelStrategy) {
+
+    if (Constants::SHOW_BALL_HANDLE_DEBUG_INFO()) {
+        std::stringstream ss;
+        ss << "travel with strategy: ";
+        switch (travelStrategy) {
+        case forwards:ss << "forwards";
+            break;
+        case backwards:ss << "backwards";
+            break;
+        case defaultTravel:ss << "default";
+            break;
+        }
+        std::cout << ss.str() << std::endl;
+    }
+
+    switch (travelStrategy) {
+    case backwards: {
+        updateBackwardsProgress();
+
+        switch (backwardsProgress) {
+        case B_start: return B_startTravelBackwards();
+        case B_turning:return B_sendTurnCommand();
+        case B_approaching:return B_sendApproachCommand();
+        case B_overshooting:return B_sendOvershootCommand();
+        case B_dribbling:return B_sendDribblingCommand();
+        case B_dribbleBackwards:return B_sendDribbleBackwardsCommand();
+        case B_success:return B_sendSuccessCommand();
+        case B_fail: {
+            backwardsProgress = B_start;
+            return {};
+        }
+        }
+    }
+    case forwards: {
+        updateForwardsProgress();
+
+        switch (forwardsProgress) {
+        case F_start:return F_startTravelForwards();
+        case F_turning:return F_sendTurnCommand();
+        case F_approaching:return F_sendApproachCommand();
+        case F_dribbleForward:return F_sendDribbleForwardsCommand();
+        case F_success:return F_sendSuccessCommand();
+        case F_fail: {
+            forwardsProgress = F_start;
+            return {};
+        }
+        }
+    }
+
+    case defaultTravel:break;
+    }
+
+    std::cout << "travel case not handled" << std::endl;
```

Comment:
This print shouldn't be necessary as compiler should explicitly warn you in this case. Can do no big harm though.